tremor-script/src/lexer.rs

```diff
@@ -2423,8 +2434,8 @@ impl<'input> Lexer<'input> {
             Some((_, 'x')) => {
                 self.bump(); // Skip 'x'
                 let int_start = self.next_index()?;
-                let (end, hex) = self.take_while(int_start, is_hex);
-                match int {
+                let (end, hex) = self.extract_number(int_start, is_hex);
+                match &int[..] {
```

Comment:
We generally forbid `[]` in the codebase since there are many pitfalls to it (read: it can panic). However this use isn't an issue you can mark it as 'sane' by adding:

```rust
// ALLOW: this takes the whole since and can not panic
```

in the line before it (same goes for the one in 2467) that will hopefully satisfy the CI gods :D